api/server/router/container/backend.go

```diff
@@ -0,0 +1,74 @@
+package container
+
+import (
+	"io"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/daemon"
+	"github.com/docker/docker/daemon/exec"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/version"
+	"github.com/docker/docker/runconfig"
+)
+
+// execBackend includes functions to implement to provide exec functionality.
+type execBackend interface {
+	ContainerExecCreate(config *runconfig.ExecConfig) (string, error)
+	ContainerExecInspect(id string) (*exec.Config, error)
+	ContainerExecResize(name string, height, width int) error
+	ContainerExecStart(name string, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) error
+	ExecExists(name string) (bool, error)
+}
+
+// copyBackend includes functions to implement to provide container copy functionality.
+type copyBackend interface {
+	ContainerArchivePath(name string, path string) (content io.ReadCloser, stat *types.ContainerPathStat, err error)
+	ContainerCopy(name string, res string) (io.ReadCloser, error)
+	ContainerExport(name string, out io.Writer) error
+	ContainerExtractToDir(name, path string, noOverwriteDirNonDir bool, content io.Reader) error
+	ContainerStatPath(name string, path string) (stat *types.ContainerPathStat, err error)
+}
+
+// stateBackend includes functions to implement to provide container state lifecycle functionality.
+type stateBackend interface {
+	ContainerCreate(params *daemon.ContainerCreateConfig) (types.ContainerCreateResponse, error)
+	ContainerKill(name string, sig uint64) error
+	ContainerPause(name string) error
+	ContainerRename(oldName, newName string) error
+	ContainerResize(name string, height, width int) error
+	ContainerRestart(name string, seconds int) error
+	ContainerRm(name string, config *daemon.ContainerRmConfig) error
+	ContainerStart(name string, hostConfig *runconfig.HostConfig) error
+	ContainerStop(name string, seconds int) error
+	ContainerUnpause(name string) error
+	ContainerWait(name string, timeout time.Duration) (int, error)
+	Exists(id string) bool
+	IsPaused(id string) bool
+}
+
+// monitorBackend includes functions to implement to provide containers monitoring functionality.
+type monitorBackend interface {
+	ContainerChanges(name string) ([]archive.Change, error)
+	ContainerInspect(name string, size bool, version version.Version) (interface{}, error)
```

Comment:
`interface{}` is the most abstract representation of a struct in Go. It basically means "whatever". It's usage is usually discouraged, but we can return three different types in that function.

A better implementation would be to return a real interface that all three struct fulfill, but since those struct only hold information in fields I don't really know if there is anything better than returning `interface{}`.